### PR TITLE
GitHub App support

### PR DESCRIPTION
**Pull request by quvide**
Adds GitHub App integration. This enables Feedbacker to access private repositories if developers authorize the app to read their contents.

## Caveats / Security
Currently the access token for the installation is exposed in the git clone url envvar, which is then echoed in the instance logs. If someone sees the access token, they can use it to access private data.

## API changes
A whole new class of endpoints exist under `/api/github/`. Some relate to Oauth2 and some are used to fetch data retrieved from the GitHub API.

## UI changes
* New tab switcher in the create view. It is used to switch between choosing a repo after authenticating with GitHub, and the other is the old basic url input.

2019-03-21-081528_644x264_scrot

## Bugs
* (Probably a lot of missing error handling in both frontend and backend)

----

## Review Checklist

- [x] All changes are described in the PR description
- [x] New feature and breaking changes have documentation
- [x] No unnecessary or debugging code
- [x] Questionable code is clearly marked with comments
- [x] New endpoints have API tests
- [x] New frontend features have Enzyme tests
- [x] Test setup on some other computer than Elias'

----

## Issues

- [x] Design bug for button
- [x] Only works for node apps, hardcoded?
- [x] Tokens to DB
- [x] Disabled when fetching to dropdowns
- [x] Logout
- [x] Multiuser bug?
- [x] Better documentation, Samuli couldn't set up on the current docs

### CODE REVIEW

File: client/src/dev-components/create/create.js [hunk 1]
```diff
@@ -116,6 +153,63 @@ class Create extends React.Component {
...
+      return (<p>Loading GitHub login information...</p>)
+    }
+
+    if (this.state.github && this.state.github.status) {
+      return (
+        <div className="github_integration">
+          <p>You are logged in to GitHub as {this.state.github.status.login}.</p>
+          <label htmlFor="installation">
+            Please select an installation
+            <select defaultValue="-1" name="installation" id="installation" form="form" onChange={this.selectedInstallationChanged}>
```
Comment (joonalillfors) [inline client/src/dev-components/create/create.js:200]: fix long lines like this to multilines?

Comment (quvide) [reply client/src/dev-components/create/create.js:200]: maybe when create.js is refactored completely

**Review by bqqbarbhg — APPROVED** [review]
FINAL SHIP IT

**Comment by quvide** [comment]
SHIP IT :rocket:

**Pull request merged**